src/Raven.Server/Documents/ETL/Providers/AI/Embeddings/EmbeddingsGenerationTask.cs

```diff
@@ -0,0 +1,487 @@
+    {
+        return new DocumentsToEmbeddingsGenerationItems(docs, collection);
+    }
+
+    protected override IEnumerator<EmbeddingsGenerationItem> ConvertTombstonesEnumerator(DocumentsOperationContext context, IEnumerator<Tombstone> tombstones, string collection,
+        bool trackAttachments)
+    {
+        return new TombstonesToEmbeddingsGenerationItems(tombstones, collection);
+    }
+
+    protected override IEnumerator<EmbeddingsGenerationItem> ConvertAttachmentTombstonesEnumerator(DocumentsOperationContext context, IEnumerator<Tombstone> tombstones,
+        List<string> collections)
+    {
+        throw new NotSupportedException($"{nameof(ConvertAttachmentTombstonesEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<EmbeddingsGenerationItem> ConvertCountersEnumerator(DocumentsOperationContext context, IEnumerator<CounterGroupDetail> counters,
+        string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertCountersEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<EmbeddingsGenerationItem> ConvertTimeSeriesEnumerator(DocumentsOperationContext context, IEnumerator<TimeSeriesSegmentEntry> timeSeries,
+        string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertTimeSeriesEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<EmbeddingsGenerationItem> ConvertTimeSeriesDeletedRangeEnumerator(DocumentsOperationContext context,
+        IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertTimeSeriesDeletedRangeEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override EtlTransformer<EmbeddingsGenerationItem, EmbeddingGenerationScriptResult, EmbeddingsGenerationStatsScope, EmbeddingsGenerationPerformanceOperation>
+        GetTransformer(DocumentsOperationContext context)
+    {
+        return new EmbeddingsGenerationScriptTransformer(Database, context, Transformation, null, Configuration);
+    }
+
+    protected override string LoadFailureMessage => $"Failed to generate embeddings in '{Configuration.Name}' task. Going to do the retry using '{Database.Configuration.Ai.EmbeddingsGenerationTaskRetryStrategy}' backoff strategy.";
+
+    protected override void EnterFallbackMode(Exception e, DateTime? lastErrorTime)
+    {
+        _fallbackCounter++;
+        var taskRetryDelay = Database.Configuration.Ai.EmbeddingsGenerationTaskRetryDelay.AsTimeSpan;
+        if (lastErrorTime == null)
+        {
+            FallbackTime = taskRetryDelay;
+        }
+        else
+        {
+            var secondsToWait = (Database.Configuration.Ai.EmbeddingsGenerationTaskRetryStrategy) switch
```

Comment:
This is a lot more complex than it needs to be. Sometimes, the model will give us the next time to check.
See: `Retry-After: 60` or `Retry-After: Thu, 13 Mar 2025 12:00:00 GMT`